src/control/server/server_utils_test.go

```diff
@@ -555,6 +556,35 @@ func TestServer_prepBdevStorage(t *testing.T) {
 			expMemSize:      16384,
 			expHugePageSize: 2,
 		},
+		"4 engines; hugepage alloc across numa 0,1": {
```

Comment:
Would be interesting to add a test for the case where there is an uneven distribution of engines.

Reply:
On inspection decided that it should be an error if engine configs are imbalanced across numa nodes (unless all engines are on a single node). Added unit test case for uneven distribution as requested.